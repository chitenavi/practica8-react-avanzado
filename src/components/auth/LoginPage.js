import React from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'antd';
import MainLayout from '../layout/MainLayout';
import Loader from '../shared/LoaderStyled';
import Button from '../shared/Button';
import { login } from '../../api/auth';

import './LoginPage.scss';

class LoginPage extends React.Component {
  state = {
    form: {
      email: '',
      password: '',
      remcredentials: false,
    },
    submitting: false,
    error: null,
  };

  canSubmit = () => {
    const {
      form: { email, password },
      submitting,
    } = this.state;
    return !submitting && email && password;
  };

  handleSubmit = async event => {
    event.preventDefault();

    this.setState({ submitting: true, error: null });
    const { onLogin, history } = this.props;
    const { form } = this.state;

    try {
      const { token } = await login(form);

      this.setState({ error: null });
      onLogin(token).then(() => history.push('/adverts'));
    } catch (error) {
      this.setState({ error });
    } finally {
      this.setState({ submitting: false });
    }
  };

  handleChange = ev => {
    this.setState(state => ({
      form: { ...state.form, [ev.target.name]: ev.target.value },
    }));
  };

  handleChangeCheck = ev => {
    this.setState(state => ({
      form: { ...state.form, [ev.target.name]: ev.target.checked },
    }));
  };

  render() {
    const {
      form: { email, password, remcredentials },
      error,
      submitting,
    } = this.state;
    return (
      <MainLayout title="Welcome to Nodepop SPA">
        <div className="loginPage">
          <form onSubmit={this.handleSubmit} className="formLogin">
            <div className="formLogin-field">
              <input
                type="text"
                onChange={this.handleChange}
                name="email"
                value={email}
                placeholder="email"
              />
            </div>
            <div className="formLogin-field">
              <input
                type="password"
                onChange={this.handleChange}
                value={password}
                name="password"
                placeholder="password"
              />
            </div>

            <div className="formLogin-field">
              <label htmlFor="remember">
                <input
                  type="checkbox"
                  id="remember"
                  name="remcredentials"
                  onChange={this.handleChangeCheck}
                  checked={remcredentials}
                />
                Remember credentials
              </label>
            </div>
            <div className="formLogin-field">
              <Button
                type="submit"
                className="secondary"
                disabled={!this.canSubmit()}
              >
                Log in
              </Button>
            </div>
          </form>
          {submitting && (
            <div className="loginPage-loading">
              <Loader size="medium" />
            </div>
          )}
          {error && (
            <div className="loginPage-error">
              <Alert message={error.message} type="error" />
            </div>
          )}
        </div>
      </MainLayout>
    );
  }
}

LoginPage.propTypes = {
  onLogin: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default LoginPage;
