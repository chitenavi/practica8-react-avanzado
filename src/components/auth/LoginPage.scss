@import '../../scss/variables';

.loginPage {
  margin: 2rem auto 0 auto;
  width: 350px;

  &-error {
    margin-top: 1rem;
    text-align: center;
  }
}

.formLogin {
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 1rem;
  align-items: center;

  padding: 2rem 1rem;
  background-color: $bg-color-ter;
  box-shadow: $shadow-primary;
  border-radius: 1rem;

  &-field {
    width: 300px;

    & button {
      width: 100%;
    }

    & input[type='text'],
    & input[type='password'] {
      height: 2rem;
      width: 100%;
      border-radius: 0.5rem;
      font-family: inherit;
      font-size: 1.2rem;
      color: black;
    }
  }
}
