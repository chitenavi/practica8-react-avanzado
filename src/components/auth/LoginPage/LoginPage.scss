@import '../../../scss/variables';

.loginPage {
  margin: 2rem auto 0 auto;
  width: 350px;

  &-error {
    margin-top: 1rem;
    text-align: center;
  }

  & .form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    gap: 1rem;
    align-items: center;

    padding: 2rem 1rem;
    background-color: $bg-color-ter;
    box-shadow: $shadow-primary;
    border-radius: 1rem;

    & .form-field {
      width: 300px;

      & .checkbox {
        margin-right: 0.3rem;
      }
    }
  }
}
