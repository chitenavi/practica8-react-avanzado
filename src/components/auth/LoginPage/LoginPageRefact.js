import React from 'react';

import MainLayout from '../../layout/MainLayout';
import { Form, InputCustom } from '../../shared/Form';

function LoginPageRefact() {
  const formLogin = {
    email: '',
    password: '',
    tags: [],
    price: [1, 10000],
  };

  // const canSubmit = () => {
  //   return !loading && email && password;
  // };

  const handleSubmit = data => {
    // const credentials = form;
    // ev.preventDefault();
    console.log(data);
    // console.log(formLogin);
  };

  return (
    <MainLayout title="Welcome to Nodepop SPA">
      <div>
        <Form
          initialValue={formLogin}
          onSubmit={handleSubmit}
          submitLabel="Log In"
        >
          <InputCustom type="text" name="email" />
          <InputCustom type="password" name="password" />
          <InputCustom type="select" name="tags" />
          <InputCustom type="slider" name="price" />
        </Form>
      </div>
    </MainLayout>
  );
}

export default LoginPageRefact;
