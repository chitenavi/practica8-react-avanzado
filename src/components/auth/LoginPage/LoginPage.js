import React from 'react';
import PropTypes from 'prop-types';

import MainLayout from '../../layout/MainLayout';
import ModalLoader from '../../shared/ModalLoader';
import ErrorMessage from '../../errors/ErrorMessage';

import Form, { InputCustom, CheckboxCustom } from '../../shared/Form';
import { formLogin } from '../../../config';

import './LoginPage.scss';

function LoginPage({ onLogin, loading, error }) {
  const handleSubmit = data => {
    onLogin(data);
  };

  return (
    <MainLayout title="Welcome to Nodepop SPA">
      <div className="loginPage">
        <Form
          initialValue={formLogin}
          onSubmit={handleSubmit}
          submitLabel="Log In"
        >
          <InputCustom type="email" name="email" placeholder="Your email" />
          <InputCustom
            type="password"
            name="password"
            placeholder="Your password"
          />
          <CheckboxCustom label="Remember Credentials" name="remcredentials" />
        </Form>
        {loading && <ModalLoader />}
        {error && (
          <ErrorMessage className="loginPage-error" message={error.message} />
        )}
      </div>
    </MainLayout>
  );
}

LoginPage.propTypes = {
  onLogin: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.objectOf(PropTypes.any),
};

LoginPage.defaultProps = {
  error: null,
};

export default LoginPage;
