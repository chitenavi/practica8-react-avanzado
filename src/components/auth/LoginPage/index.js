import { connect } from 'react-redux';

import LoginPage from './LoginPage';
import { login } from '../../../store/actions';
import { getUi } from '../../../store/selectors';

const mapStateToProps = getUi;
const mapDispatchToProps = dispatch => ({
  onLogin: credentials => dispatch(login(credentials)),
});

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
