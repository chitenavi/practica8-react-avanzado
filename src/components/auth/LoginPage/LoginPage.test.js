import React from 'react';
import { shallow } from 'enzyme';

import LoginPage from './LoginPage';

describe('LoginPage', () => {
  const onLogin = jest.fn();
  const loading = false;
  const error = null;

  const render = () =>
    shallow(<LoginPage onLogin={onLogin} loading={loading} error={error} />);
  it('snapshot', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });
  it('snapshot with error', () => {
    const wrapper = render();
    wrapper.setProps({ error: { message: 'Error' } });
    expect(wrapper.exists('.loginPage-error')).toBe(true);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call a onLogin', () => {
    const data = {};
    const wrapper = render();

    const form = wrapper.find('Form');
    form.simulate('submit', data);

    expect(wrapper.exists()).toBe(true);
    expect(onLogin).toHaveBeenCalled();
  });
});
