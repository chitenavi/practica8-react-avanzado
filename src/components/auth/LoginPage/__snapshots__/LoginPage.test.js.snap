// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage snapshot 1`] = `
<MainLayout
  title="Welcome to Nodepop SPA"
>
  <div
    className="loginPage"
  >
    <Form
      encType="application/x-www-form-urlencoded"
      initialValue={
        Object {
          "email": "",
          "password": "",
          "remcredentials": false,
          "validateFields": Array [
            "email",
            "password",
          ],
        }
      }
      method="GET"
      onSubmit={[Function]}
      submitLabel="Log In"
    >
      <InputCustom
        className=""
        name="email"
        placeholder="Your email"
        type="email"
      />
      <InputCustom
        className=""
        name="password"
        placeholder="Your password"
        type="password"
      />
      <CheckboxCustom
        className=""
        label="Remember Credentials"
        name="remcredentials"
      />
    </Form>
  </div>
</MainLayout>
`;

exports[`LoginPage snapshot with error 1`] = `
<MainLayout
  title="Welcome to Nodepop SPA"
>
  <div
    className="loginPage"
  >
    <Form
      encType="application/x-www-form-urlencoded"
      initialValue={
        Object {
          "email": "",
          "password": "",
          "remcredentials": false,
          "validateFields": Array [
            "email",
            "password",
          ],
        }
      }
      method="GET"
      onSubmit={[Function]}
      submitLabel="Log In"
    >
      <InputCustom
        className=""
        name="email"
        placeholder="Your email"
        type="email"
      />
      <InputCustom
        className=""
        name="password"
        placeholder="Your password"
        type="password"
      />
      <CheckboxCustom
        className=""
        label="Remember Credentials"
        name="remcredentials"
      />
    </Form>
    <ErrorMessage
      className="loginPage-error"
      message="Error"
    />
  </div>
</MainLayout>
`;
